pmemcheck/pmc_tx.c

```diff
@@ -423,12 +440,20 @@ add_obj_to_tx(UWord tx_id, UWord base, UWord size)
         /* omit self */
         if (tx_iter->tx_id == tx_id)
             continue;
-        if (is_in_mapping_set(&reg, tx_iter->regions))
+        if (is_in_cache(&reg, tx_iter))
```

Comment:
It may happen that *cached_region* includes *reg* but they do not match exactly. Seems like `cmp_pmem_st()` should be used here.